src/diffusers/pipelines/blip_diffusion/modeling_blip2.py

```diff
@@ -0,0 +1,657 @@
+from transformers.activations import QuickGELUActivation as QuickGELU
+from transformers.models.blip_2.modeling_blip_2 import Blip2Attention, Blip2MLP, Blip2EncoderLayer, Blip2Encoder, Blip2QFormerAttention, Blip2QFormerIntermediate, Blip2QFormerOutput, Blip2PreTrainedModel
+
+logger = logging.get_logger(__name__)
+
+""" PyTorch BLIP-2 model.
+    There is an implementation of Blip2 in Transformers : https://github.com/huggingface/transformers/blob/main/src/transformers/models/blip_2/modeling_blip_2.py
+    Which doesn't support getting multimodal embeddings for now
+    Will replace this with the Transformers version once it supports multimodal embeddings 
+"""
+
+
+class Blip2TextEmbeddings(nn.Module):
+    """Construct the embeddings from word and position embeddings."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.word_embeddings = nn.Embedding(
+            config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id
+        )
+        self.position_embeddings = nn.Embedding(
+            config.max_position_embeddings, config.hidden_size
+        )
+
+        # self.LayerNorm is not snake-cased to stick with TensorFlow model variable name and be able to load
+        # any TensorFlow checkpoint file
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+        # position_ids (1, len position emb) is contiguous in memory and exported when serialized
+        self.register_buffer(
+            "position_ids", torch.arange(config.max_position_embeddings).expand((1, -1))
+        )
+        self.position_embedding_type = getattr(
+            config, "position_embedding_type", "absolute"
+        )
+
+        self.config = config
+
+    def forward(
+        self,
+        input_ids=None,
+        position_ids=None,
+        query_embeds=None,
+        past_key_values_length=0,
+    ):
+        if input_ids is not None:
+            seq_length = input_ids.size()[1]
+        else:
+            seq_length = 0
+
+        if position_ids is None:
+            position_ids = self.position_ids[
+                :, past_key_values_length : seq_length + past_key_values_length
+            ].clone()
+
+        if input_ids is not None:
+            embeddings = self.word_embeddings(input_ids)
+            if self.position_embedding_type == "absolute":
+                position_embeddings = self.position_embeddings(position_ids)
+                embeddings = embeddings + position_embeddings
+
+            if query_embeds is not None:
+                embeddings = torch.cat((query_embeds, embeddings), dim=1)
+        else:
+            embeddings = query_embeds
+        embeddings = self.LayerNorm(embeddings)
+        embeddings = self.dropout(embeddings)
+        return embeddings
+
+
+
+# Copied from transformers.models.blip.modeling_blip.BlipVisionEmbeddings with Blip->Blip2
```

Comment:
This "Copied from ...` comments (applicable for the comments that have `transformers` mentions) will break our CI since there's no local module named `transformers`. So, maybe let's just stick to `# Copy-pasted ...`.